test/integration/executor/set_account_detail_test.cpp

```diff
@@ -0,0 +1,133 @@
+  iroha::ametsuchi::CommandResult setDetail(const AccountIdType &target,
+                                            const AccountDetailKeyType &key,
+                                            const AccountDetailValueType &value,
+                                            const AccountIdType &issuer) {
+    return getItf().executeCommandAsAccount(
+        *getItf().getMockCommandFactory()->constructSetAccountDetail(
+            target, key, value),
+        issuer,
+        true);
+  }
+
+  void checkDetails(AccountIdType account,
+                    DetailsByKeyByWriter reference_details) {
+    assertResultValue(
+        getItf()
+            .executeQueryAndConvertResult(
+                *getItf().getMockQueryFactory()->constructGetAccountDetail(
+                    account, boost::none, boost::none, boost::none))
+            .specific_response
+        | [&reference_details](const auto &response) {
+            checkJsonData(response.detail(), reference_details);
+            return iroha::expected::Value<void>{};
+          });
+  }
+};
+
+/**
+ * C274
+ * @given a user without can_set_detail permission
+ * @when execute SetAccountDetail command to set own detail
+ * @then the command succeeds and the detail is added
+ */
+TEST_P(SetAccountDetailTest, Self) {
+  getItf().createUserWithPerms(kUser, kDomain, kUserKeypair.publicKey(), {});
+  assertResultValue(setDetail(kUserId, kKey, kVal, kUserId));
+  checkDetails(kUserId, DetailsByKeyByWriter{{{kUserId, {{kKey, kVal}}}}});
+}
+
+/**
+ * C273
+ * @given a user with all required permissions
+ * @when execute SetAccountDetail command with nonexistent user
+ * @then the command fails with error code 3
+ */
+TEST_P(SetAccountDetailTest, NonExistentUser) {
+  checkCommandError(setDetail(kUserId, kKey, kVal, kAdminId), 3);
+}
+
+/**
+ * C280
+ * @given a pair of users and first one without permissions
+ * @when the first one tries to execute SetAccountDetail on the second
+ * @then the command does not succeed and the detail is not added
+ */
+TEST_P(SetAccountDetailTest, NoPerms) {
+  getItf().createUserWithPerms(kUser, kDomain, kUserKeypair.publicKey(), {});
+  getItf().createUserWithPerms(
+      kAnotherUser, kDomain, kSameDomainUserKeypair.publicKey(), {});
+  assertResultError(setDetail(kSameDomainUserId, kKey, kVal, kUserId));
+  checkDetails(kSameDomainUserId, DetailsByKeyByWriter{});
+}
+
+/**
+ * @given a pair of users and first one has can_set_detail permission
+ * @when the first one executes SetAccountDetail on the second
+ * @then the command succeeds and the detail is added
+ */
+TEST_P(SetAccountDetailTest, ValidRolePerm) {
+  getItf().createUserWithPerms(
+      kUser, kDomain, kUserKeypair.publicKey(), {Role::kSetDetail});
+  getItf().createUserWithPerms(
+      kAnotherUser, kDomain, kSameDomainUserKeypair.publicKey(), {});
+  assertResultValue(setDetail(kUserId, kKey, kVal, kUserId));
+  checkDetails(kUserId, DetailsByKeyByWriter{{{kUserId, {{kKey, kVal}}}}});
```

Comment:
`}}}}}` looks strong

Reply:
I was inspired by the double `+` in c`c++`